pkg/model/trace/combine.go

```diff
@@ -129,19 +142,26 @@ func (c *Combiner) ConsumeWithFinal(tr *tempopb.Trace, final bool) (int, error)
 	}
 
 	c.combined = true
-	return spanCount, c.sizeError()
+	moveForward, maxSizeErr := c.canContinue()
+	if moveForward {
+		return spanCount, nil
+	}
+
+	return spanCount, maxSizeErr
 }
 
-func (c *Combiner) sizeError() error {
+func (c *Combiner) canContinue() (bool, error) {
```

Comment:
i don't understand this "move forward" bool. it seems like this function signature should stay the same.

Reply:
This is my answer to this: https://github.com/grafana/tempo/pull/3941#pullrequestreview-2224443581 
To be honest, both approaches are fine to me. Looks more like a stylistic thing

Reply:
Making `canContinue()` return a bool but keeping it in the combiner is not worth it. We need to either:

- Return the bool out of the combiner so the calling code can stop looping over traces to combine
- Restore it to just return an error for simplicity